### PR TITLE
Using SDK MethodInvoker rather than reflection for method invoke

### PR DESCRIPTION
**Pull request by mathewc**

### CODE REVIEW

File: test/WebJobs.Script.Tests.Integration/WebHostEndToEnd/ProxyEndToEndTests.cs [hunk 1]
```diff
@@ -28,11 +28,13 @@ public class ProxyEndToEndTests : IClassFixture<ProxyEndToEndTests.TestFixture>
     {
         private readonly ScriptSettingsManager _settingsManager;
         private TestFixture _fixture;
+        private string _hostName;
 
         public ProxyEndToEndTests(TestFixture fixture)
         {
             _fixture = fixture;
             _settingsManager = ScriptSettingsManager.Instance;
+            _hostName = new HostNameProvider(SystemEnvironment.Instance).Value ?? "localhost";
```
Comment (mathewc) [inline test/WebJobs.Script.Tests.Integration/WebHostEndToEnd/ProxyEndToEndTests.cs:37]: Often on my machine I have a WEBSITE_SITE_NAME env var set which breaks these tests.

File: src/WebJobs.Script/Description/DotNet/DotNetFunctionInvoker.cs [hunk 2]
```diff
@@ -267,7 +269,7 @@ protected override async Task<object> InvokeCore(object[] parameters, FunctionIn
             int actualParameterCount = function.GetParameters().Length;
             parameters = parameters.Take(actualParameterCount).ToArray();
 
-            object result = function.Invoke(null, parameters);
+            object result = await _methodInvoker.InvokeAsync(null, parameters);
```
Comment (mathewc) [inline src/WebJobs.Script/Description/DotNet/DotNetFunctionInvoker.cs:272]: Replacing reflection invoke with call to compiled lambda. We're now using the same MethodInvokerFactory that the core SDK uses when invoking methods.

File: .nuget/NuGet.Config [hunk 3]
```diff
@@ -3,6 +3,7 @@
   <packageSources>
     <add key="nuget.org" value="https://www.nuget.org/api/v2/" />
     <add key="azure_app_service" value="https://www.myget.org/F/azure-appservice/api/v2" />
+    <add key="azure-appservice-test" value="https://azfunc.pkgs.visualstudio.com/e6a70c92-4128-439f-8012-382fe78d6396/_packaging/azure-appservice-test%40Local/nuget/v3/index.json" />
```
Comment (mathewc) [inline .nuget/NuGet.Config:6]: Adding new test package source I created in devops for intermediate test packages. E.g. I've uploaded packages that contain new SDK APIs this PR relies on, which allows the PR build to pass, before those changes are actually published.

File: src/WebJobs.Script.WebHost/WebJobs.Script.WebHost.csproj [hunk 4]
```diff
@@ -64,7 +64,7 @@
     <PackageReference Include="Microsoft.Azure.Services.AppAuthentication" Version="1.0.3" />
     <PackageReference Include="Microsoft.Azure.Storage.File" Version="11.1.7" />
     <PackageReference Include="Microsoft.Azure.WebJobs.Host.Storage" Version="4.0.1" />    
-    <PackageReference Include="Microsoft.Azure.WebJobs" Version="3.0.20" />
+    <PackageReference Include="Microsoft.Azure.WebJobs" Version="3.0.21-dev637336297117359276" />
```
Comment (mathewc) [inline src/WebJobs.Script.WebHost/WebJobs.Script.WebHost.csproj]: Temp - will update with actual package version

File: src/WebJobs.Script/Description/DotNet/DotNetFunctionInvoker.cs [hunk 5]
```diff
@@ -276,12 +278,6 @@ protected override async Task<object> InvokeCore(object[] parameters, FunctionIn
                 originalParameters[i] = parameters[i];
             }
 
-            // unwrap the task
-            if (result is Task)
-            {
-                result = await ((Task)result).ContinueWith(t => GetTaskResult(t), TaskContinuationOptions.ExecuteSynchronously);
```
Comment (mathewc) [inline src/WebJobs.Script/Description/DotNet/DotNetFunctionInvoker.cs:282]: No longer need to use reflection to unwrap the task result

**Review by brettsam — DISMISSED** [review]

**Comment by brettsam** [comment]
I'll have to re-approve once you update the nugets.

**Review by brettsam — APPROVED** [review]

**Review by paulbatum — APPROVED** [review]

**Pull request merged**